src/main/scala/com/snowflake/snowpark/functions.scala

```diff
@@ -3140,6 +3140,112 @@ object functions {
    */
   def listagg(col: Column): Column = listagg(col, "", isDistinct = false)
 
+  /**
+   * This function receives a column and extracts the groupIdx from the string
+   * after applying the exp regex. Returns empty string when the string doesn't
+   *  match and null if the input is null.
+   *
+   * This function applies the `case sensitive` and `extract` flags.
+   * It doesn't apply multiline nor .* matches newlines.
+   * If these flags need to be applied, use `builtin("REGEXP_SUBSTR")`
```

Comment:
maybe we should expose REGEX_SUBSTR as part of this PR, just to avoid using builtin("REGEXP_SUBSTR") which will be later deprecated

Reply:
modified